### PR TITLE
docs: fix links and badges error

### LINKED ISSUE #5238: Badges in docs breaks link's references
### Describe the bug

If exists a badge in a title the html id will be `title` + `badge information` and breaks the links that reference this title.

For example, in docs there are several reference to `/config/#browser-fileparallelism` but this doesn't exist, instead it should be `/config/#browser-fileparallelism-1-3-0`

Solution: add specific id in the markdown

```md
### browser.fileParallelism <Badge type="info">1.3.0+</Badge> {#browser-fileparallelism}
```

### Reproduction

image

### System Info

```shell
browser
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by wokcito**
fix #5238

### Description

Fixed links that weren't working in documentation and badge error specified in the issue #5238

For example the link `vi.fn()` [here](https://vitest.dev/api/vi.html#vi-spyon) redirects to the main page, it is now fixed

I checked all links in documentation

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [ ] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: docs/guide/ui.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Then you can start the tests with UI by passing the `--ui` flag:
 vitest --ui
 ```
 
-Then you can visit the Vitest UI at <a href="http://localhost:51204/__vitest__/">`http://localhost:51204/__vitest__/`</a>
+Then you can visit the Vitest UI at [`http://localhost:51204/__vitest__/`](http://localhost:51204/__vitest__/)
```
Comment (sheremet-va) [inline docs/guide/ui.md:19]: This breaks docs build

Comment (wokcito) [reply docs/guide/ui.md:19]: Sorry for that, should I make other PR, other commit fixing it or force the push with the change?

Comment (wokcito) [reply docs/guide/ui.md:19]: Sorry about that. Should I make other PR, commit or forze a push with the change?

Comment (wokcito) [reply docs/guide/ui.md]: Sorry about that. I revert that change.

**Pull request merged**